### PR TITLE
Fix return flows to always flow downstream

### PR DESCRIPTION
**Pull request by jrising**
This is the optimization-side fix for return flow problem.  It specifies that return flows are driven entirely based on withdrawals, but with the percentages based on the county's recorded demands.

### CODE REVIEW

File: src/WaterNetwork.jl [hunk 1]
```diff
@@ -55,23 +55,38 @@ function initwaternetwork(m::Model)
...
+    for hh in 1:numgauges
+        gg = vertex_index(downstreamorder[hh])
+        gauge = downstreamorder[hh].label
+        for upstream in out_neighbors(wateridverts[gauge], waternet)
+            index = vertex_index(upstream, waternet)
+            A[gg, :] += A[index, :]
+        end
+    end
+end
+
```
[the remark this answers is not in the document]
Comment (jrising) [reply src/WaterNetwork.jl]: Good call.  I'm actually going to rename `src/waternet.jl` to `initwaternet.jl` and add a new `src/lib/waternet.jl`.  That way, the files directly in `src/` are all either components or scripts, while useful helper functions are defined in `src/lib/`.

File: src/WaterNetwork.jl [hunk 2]
```diff
@@ -55,23 +55,38 @@ function initwaternetwork(m::Model)
...
-            gaugeid = draws[pp, :gaugeid]
-            vertex = get(wateridverts, gaugeid, nothing)
-            if vertex == nothing
-                println("Missing $gaugeid")
-            else
-                gg = vertex_index(vertex)
-                A[gg, pp] = -1.
-            end
-        end
+        matrix_gauges_canals(A, -.99 * ones(nrow(draws)))
```
[the remark this answers is not in the document]
Comment (jrising) [reply src/WaterNetwork.jl]: This is a placeholder for a number to represent some loss, but I realize now that it actually allows 1% additional flow to come out of the river.  I put these parameters as constants into the new `src/lib/waternet.jl`.

**Comment by jrising** [comment]
@LaurelineJ I've also implemented simulation of return flows here.

**Comment by jrising** [comment]
@LaurelineJ Any objections to me merging this into master?  The unit tests fail for reasons that appear unique to the Travis CI system, resulting in a slightly different optimization.  I've tried things to fix it, and now am inclined to ignore it.

**Comment by LaurelineJ** [comment]
Sorry I am seeing only this now, but of course yes!

**Pull request merged**